modin/experimental/engines/omnisci_on_native/test/utils.py

```diff
@@ -0,0 +1,195 @@
+            else:
+                raise TypeError(
+                    f"Unexpected storage format, expected pandas.DataFrame or pyarrow.Table, got: {type(frame)}."
+                )
+            partition.frame_id = frame_id
+            self._imported_frames.append((df, frame_id))
+
+    def __enter__(self, *dfs):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        for df, frame_id in self._imported_frames:
+            actual_frame_id = df._query_compiler._modin_frame._partitions[0][0].frame_id
+            OmnisciServer().executeDDL(f"DROP TABLE IF EXISTS {frame_id}")
+            if actual_frame_id == frame_id:
+                df._query_compiler._modin_frame._partitions[0][0].frame_id = None
+        self._imported_frames = []
+
+
+def set_execution_mode(frame, mode, recursive=False):
+    if isinstance(frame, (pd.Series, pd.DataFrame)):
+        frame = frame._query_compiler._modin_frame
+    frame._force_execution_mode = mode
+    if recursive and hasattr(frame._op, "input"):
+        for child in frame._op.input:
+            set_execution_mode(child, mode, True)
+
+
+def run_and_compare(
+    fn,
+    data,
+    data2=None,
+    force_lazy=True,
+    force_arrow_execute=False,
+    allow_subqueries=False,
+    comparator=df_equals,
+    **kwargs,
+):
+    def run_modin(
+        fn,
+        data,
+        data2,
+        force_lazy,
+        force_arrow_execute,
+        allow_subqueries,
+        constructor_kwargs,
+        **kwargs,
+    ):
+        kwargs["df1"] = pd.DataFrame(data, **constructor_kwargs)
+        kwargs["df2"] = pd.DataFrame(data2, **constructor_kwargs)
+        kwargs["df"] = kwargs["df1"]
+
+        if force_lazy:
+            set_execution_mode(kwargs["df1"], "lazy")
+            set_execution_mode(kwargs["df2"], "lazy")
+        elif force_arrow_execute:
+            set_execution_mode(kwargs["df1"], "arrow")
+            set_execution_mode(kwargs["df2"], "arrow")
+
+        exp_res = fn(lib=pd, **kwargs)
+
+        if force_arrow_execute:
+            set_execution_mode(exp_res, "arrow", allow_subqueries)
+        elif force_lazy:
+            set_execution_mode(exp_res, None, allow_subqueries)
+
+        return exp_res
+
+    constructor_kwargs = kwargs.pop("constructor_kwargs", {})
+    try:
+        kwargs["df1"] = pandas.DataFrame(data, **constructor_kwargs)
+        kwargs["df2"] = pandas.DataFrame(data2, **constructor_kwargs)
+        kwargs["df"] = kwargs["df1"]
+        ref_res = fn(lib=pandas, **kwargs)
+    except Exception as e:
+        with pytest.raises(type(e)):
+            exp_res = run_modin(
+                fn=fn,
+                data=data,
+                data2=data2,
+                force_lazy=force_lazy,
+                force_arrow_execute=force_arrow_execute,
+                allow_subqueries=allow_subqueries,
+                constructor_kwargs=constructor_kwargs,
+                **kwargs,
+            )
+            _ = exp_res.index
```

Comment:
why is this needed?

Reply:
it seems that it was done in order to trigger actual computations (to get the index axis we need the frame to be computed)

BTW, I haven't changed this function as well as `set_execution_mode`, these were just copied here from the `test_dataframe.py`